app/web_ui/src/lib/stores/rag_progress_store.ts

```diff
@@ -25,45 +26,103 @@ interface RagConfigurationProgressState {
   last_started_rag_config_id: string | null
 }
 
+interface ProjectRagProgressState {
+  projects: Record<string, RagConfigurationProgressState>
```

Comment:
nit: naming. This doesn't contain projects, it contains rag_config_state_by_project_id